pkg/3scale/amp/operator/system_reconciler.go

```diff
@@ -322,6 +326,34 @@ func (r *SystemReconciler) Reconcile() (reconcile.Result, error) {
 		return reconcile.Result{RequeueAfter: 5 * time.Second}, nil
 	}
 
+	// Check if enough routes exist
+	routesExist, err := r.routesExist()
```

Comment:
why are we doing this?

Reply:
resyncRoutesJob nvm

Reply:
I don't know if this is actually good enough...
So let's say I have 100s of routes created.
I disable zync - do all my routes get deleted?

Reply:
Yes all routes (except the backend route which is created by the operator) will get deleted but I thought this was what we wanted. 

@briangallagher Was this what you were talking about the other day wrt MT RHOAM on the DeveloperSandbox?

Reply:
Yes this can give us a false positive. I'll leverage the existing `defaultRoutesReady()` in  `apimanager_status_reconciler.go` instead.

Reply:
Fixed in latest commit